### PR TITLE
WHO: Two Streams Facility

### PR DESCRIPTION
**Pull request by Northmoc**

### CODE REVIEW

File: forge-gui/res/cardsfolder/t/two_streams_facility.txt [hunk 1]
```diff
@@ -0,0 +1,17 @@
+Name:Two Streams Facility
+ManaCost:no cost
+Types:Plane Apalapucia
+T:Mode$ PlaneswalkedTo | ValidCard$ Card.Self | Execute$ TrigChoose | TriggerDescription$ Whenever you planeswalk to CARDNAME and at the beginning of the first upkeep of the game, each player chooses green anchor or red waterfall.
+T:Mode$ Phase | FirstUpkeepThisGame$ True | Execute$ TrigChoose | TriggerZones$ Command | Secondary$ True | TriggerDescription$ Whenever you planeswalk to CARDNAME and at the beginning of the first upkeep of the game, each player chooses green anchor or red waterfall.
+SVar:TrigChoose:DB$ RepeatEach | RepeatPlayers$ Player | RepeatSubAbility$ DBChoice | ClearRememberedBeforeLoop$ True
+SVar:DBChoice:DB$ GenericChoice | Defined$ Remembered | Choices$ GreenAnchor,RedWaterfall | ShowChoice$ True
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/t/two_streams_facility.txt:7]: what would happen if you do `DB$ GenericChoice | Defined$ Player | Choices$ GreenAnchor,RedWaterfall` and make the Loop Player available inside the SubAbility via ChooseGenericEffect ?

Comment (Northmoc) [reply forge-gui/res/cardsfolder/t/two_streams_facility.txt:7]: Sounds good

Comment (Hanmac) [reply forge-gui/res/cardsfolder/t/two_streams_facility.txt]: i don't know if we can do this for all Cards or if that would break some cards

Comment (Northmoc) [reply forge-gui/res/cardsfolder/t/two_streams_facility.txt]: I think leaving it as a Param gives us the flexibility to more easily tweak it later... and only break cards we add the param to 😬 

I refactored some more cards... I am scared of messing with "Eureka" or "Hypergenesis"

File: forge-gui/res/cardsfolder/a/archangel_of_strife.txt [hunk 2]
```diff
@@ -4,10 +4,9 @@ Types:Creature Angel
...
-SVar:WarChoice:DB$ Pump | Defined$ Player.IsRemembered | NoteCards$ Self | NoteCardsFor$ Strife | SpellDescription$ War
-SVar:PeaceChoice:DB$ Pump | Defined$ Player.IsRemembered | ClearNotedCardsFor$ Strife | SpellDescription$ Peace
+SVar:ChooseEach:DB$ GenericChoice | Defined$ Player | TempRemember$ Chooser | Choices$ WarChoice,PeaceChoice | ShowChoice$ ExceptSelf | SpellDescription$ As CARDNAME enters the battlefield, each player chooses war or peace.
+SVar:WarChoice:DB$ Pump | Defined$ Remembered | NoteCards$ Self | NoteCardsFor$ Strife | SpellDescription$ War
+SVar:PeaceChoice:DB$ Pump | Defined$ Remembered | ClearNotedCardsFor$ Strife | SpellDescription$ Peace
 SVar:DBRemember:DB$ Pump | RememberObjects$ Player.NotedForStrife | ImprintCards$ Self
 S:Mode$ Continuous | AffectedZone$ Battlefield | Affected$ Creature.ControlledBy Remembered | AddPower$ 3 | Description$ Creatures controlled by players who chose war get +3/+0.
 S:Mode$ Continuous | AffectedZone$ Battlefield | Affected$ Creature.!ControlledBy Remembered | AddToughness$ 3 | IsPresent$ Card.IsImprinted | Description$ Creatures controlled by players who chose peace get +0/+3.
```
Comment (Hanmac) [inline forge-gui/res/cardsfolder/a/archangel_of_strife.txt:12]: Can the static below work without Remember/Imprint?

`Creature.ControlledBy Player.NotedForStrive` ?

File: forge-gui/res/cardsfolder/t/two_streams_facility.txt (hunk 1 above)
Comment (Hanmac) [reply forge-gui/res/cardsfolder/t/two_streams_facility.txt]: we could always update them later

File: forge-gui/res/cardsfolder/a/archangel_of_strife.txt [hunk 3]
```diff
@@ -7,8 +7,7 @@ K:ETBReplacement:Other:ChooseEach
...
 SVar:WarChoice:DB$ Pump | Defined$ Remembered | NoteCards$ Self | NoteCardsFor$ Strife | SpellDescription$ War
 SVar:PeaceChoice:DB$ Pump | Defined$ Remembered | ClearNotedCardsFor$ Strife | SpellDescription$ Peace
-SVar:DBRemember:DB$ Pump | RememberObjects$ Player.NotedForStrife | ImprintCards$ Self
-S:Mode$ Continuous | AffectedZone$ Battlefield | Affected$ Creature.ControlledBy Remembered | AddPower$ 3 | Description$ Creatures controlled by players who chose war get +3/+0.
-S:Mode$ Continuous | AffectedZone$ Battlefield | Affected$ Creature.!ControlledBy Remembered | AddToughness$ 3 | IsPresent$ Card.IsImprinted | Description$ Creatures controlled by players who chose peace get +0/+3.
+S:Mode$ Continuous | AffectedZone$ Battlefield | Affected$ Creature.ControlledBy Player.NotedForStrife | AddPower$ 3 | Description$ Creatures controlled by players who chose war get +3/+0.
+S:Mode$ Continuous | AffectedZone$ Battlefield | Affected$ Creature.!ControlledBy Player.NotedForStrife | AddToughness$ 3 | Description$ Creatures controlled by players who chose peace get +0/+3.
```
[the remark this answers is not in the document]
Comment (Northmoc) [reply forge-gui/res/cardsfolder/a/archangel_of_strife.txt]: Initial testing did seem OK, but either way

[the remark this answers is not in the document]
Comment (Hanmac) [reply forge-gui/res/cardsfolder/a/archangel_of_strife.txt]: its the corner case if it somehow enters without its replacement effect, or is turned face-up.

in this case, neither War nor Peace is chosen, so no effect should apply (the old code was using imprint for this)

[the remark this answers is not in the document]
Comment (Northmoc) [reply forge-gui/res/cardsfolder/a/archangel_of_strife.txt]: Ah, I wondered what that Imprint was all about. That corner case makes sense. This version is a bit cleaner on first glance too

**Review by Hanmac — APPROVED** [review]

**Pull request merged**